go/worker/sentry/worker.go

```diff
@@ -133,16 +133,21 @@ func New(backend api.Backend, identity *identity.Identity) (*Worker, error) {
 	}
 
 	// Initialize the sentry grpc worker.
-	sentryGrpcWorker, err := workerGrpcSentry.New(identity)
+	sentryGrpcWorker, err := workerGrpcSentry.New(backend, identity)
 	if err != nil {
 		return nil, fmt.Errorf("worker/sentry: failed to create a new sentry grpc worker: %w", err)
 	}
 	w.grpcWorker = sentryGrpcWorker
 
-	// Stop in case of grpc/worker quitting.
 	go func() {
-		<-w.grpcWorker.Quit()
-		w.Stop()
+		for { // nolint: S1000
+			// Stop in case of grpc/worker quitting.
+			select {
```

Comment:
Does this change anything compared to current behaviour?

Reply:
That is a leftover from some testing, I'll clean it up before this is done.